atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1399,7 +1399,8 @@ private void commitWrites(TransactionService transactionService) {
                         .build();
                     });
         } finally {
-            timelockService.unlock(ImmutableSet.of(commitLocksToken));
+            runAndGetDurationMicros(() -> timelockService.unlock(ImmutableSet.of(commitLocksToken)),
```

Comment:
From what I can see, this returns a long... which you're never using, reporting, logging or otherwise storing. I think you need to move the massive log message into the finally block, too.

Reply:
I think we only want to do the big logging in the event of a successful tx though. Might need some nontrivial refactoring to get this right!

Actually `runAndGetDurationMicros` does have side effects that report the time to a Dropwizard metric timer, which is what I was going for. I'll... rename that as well.